<launch>
  <arg name="rgb_sub_topic" default="/camera/color/image_raw/compressed" />
  <arg name="depth_sub_topic" default="/camera/aligned_depth_to_color/image_raw" />
  <arg name="camera_info_sub_topic" default="/camera/color/camera_info" />

  <!-- Contact GraspNet Parameters -->
  <arg name="z_range" default="[0.2, 1.8]" />
  <arg name="local_regions" default="true" />
  <arg name="filter_grasps" default="true" />
  <arg name="skip_border_objects" default="true" />

  <!-- OneFormer Parameters -->
  <arg name="stretch_robot_rotate_image_90deg" default="true" />

  <node
    pkg="contact_graspnet"
    name="contact_graspnet_node"
    launch-prefix="$(find-pkg-prefix contact_graspnet)/.venv/bin/python"
    exec="python_entrypoint.py"
    output="screen"
  >
    <!-- rosnode/parameters.py contains documentation of all parameters declared
    by this node. This file may contain multiple classes. Be sure to check the
    one matching this node's name. -->
    <param name="rgb_sub_topic" value="$(var rgb_sub_topic)" />
    <param name="depth_sub_topic" value="$(var depth_sub_topic)" />
    <param name="camera_info_sub_topic" value="$(var camera_info_sub_topic)" />

    <!-- Contact GraspNet Parameters -->
    <param name="contact_graspnet.z_range" value="$(var z_range)" />
    <param name="contact_graspnet.local_regions" value="$(var local_regions)" />
    <param name="contact_graspnet.filter_grasps" value="$(var filter_grasps)" />
    <param name="contact_graspnet.skip_border_objects" value="$(var skip_border_objects)" />

    <!-- OneFormer Parameters -->
    <param name="oneformer.stretch_robot_rotate_image_90deg" value="$(var stretch_robot_rotate_image_90deg)" />
  </node>

</launch>
